clouddrift/signal.py

```diff
@@ -41,57 +43,84 @@ def analytic_transform(
     >>> zp = analytic_transform(z)
     >>> zn = analytic_transform(np.conj(z))
 
-    To specify that a periodic boundary condition should be used
+    To specify that a periodic boundary condition should be used:
     >>> x = np.random.rand(99)
     >>> z = analytic_transform(x,boundary="periodic")
 
+    To specify that the time axis is along the first axis, and apply
+    zero boundary conditions:
+    >>> x = np.random.rand(100,99)
+    >>> z = analytic_transform(x,time_axis=0,boundary="zeros")
+
     Raises
     ------
     ValueError
         If ``boundary not in ["mirror", "zeros", "periodic"]``.
     """
-    # assume unidimensional input; add dimension option
-    m0 = len(x)
+    # time_axis must be in valid range
+    if time_axis < -1 or time_axis > len(x.shape) - 1:
+        raise ValueError(
+            f"time_axis ({time_axis}) is outside of the valid range ([-1,"
+            f" {len(x.shape) - 1}])."
+        )
+
+    # Reshape the inputs to ensure the time axis is last (fast-varying)
+    # should we add a condition to skip if time_axis=-1 or time_axis == len(x.shape)-1
+    # do we need to copy x here? does it matter for memory and performance?
+    if time_axis != -1 and time_axis != len(x.shape) - 1:
+        x_ = np.swapaxes(x, time_axis, -1)
```

Comment:
if you are not using `x` I would just overwrite it.

Reply:
Not a big deal, but I think it's a good practice to leave input arguments untouched because otherwise the code may be more difficult to read and understand. For example here, `x` may have a different shape in different code sections depending on input state. For a similar example, see [this](https://github.com/Cloud-Drift/clouddrift/pull/218#discussion_r1280847570) where I made a reasoning mistake because of modifying input arguments.